contracts/lib/HashLib.sol

```diff
@@ -216,10 +178,54 @@ library HashLib {
         return a.hash();
     }
 
+    function hashEIP712Domain(EIP712Domain memory erc7739Data) internal pure returns (bytes32) {
+        return keccak256(
+            abi.encode(
+                EIP712_DOMAIN_TYPEHASH,
+                keccak256(bytes(erc7739Data.name)),
+                keccak256(bytes(erc7739Data.version)),
+                erc7739Data.chainId,
+                erc7739Data.verifyingContract
+            )
+        );
+    }
+
+    function hash(EIP712Domain calldata erc7739Data) internal pure returns (bytes32) {
+        return keccak256(
+            abi.encode(
+                EIP712_DOMAIN_TYPEHASH,
+                keccak256(bytes(erc7739Data.name)),
+                keccak256(bytes(erc7739Data.version)),
+                erc7739Data.chainId,
+                erc7739Data.verifyingContract
+            )
+        );
+    }
```

Comment:
Do we need both of them?

Reply:
@filmakarov addressed both points in the commits below